src/code_7345A0.c

```diff
@@ -1,5 +1,27 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "code_7345A0", func_802A1000_7345A0);
+ApiStatus func_802A1000_7345A0(ScriptInstance *script, s32 isInitialCall) {
+    script->varTable[0] = BATTLE_STATUS->unk_83;
+    script->varTable[1] = 15;
 
-INCLUDE_ASM(s32, "code_7345A0", func_802A1050_7345F0);
+    if (is_ability_active(Ability_BERSERKER)) {
+        script->varTable[0] = 0;
+        script->varTable[1] = 40;
+    }
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A1050_7345F0(ScriptInstance *script, s32 isInitialCall) {
+    script->varTable[0] = 0;
+
+    if (is_ability_active(Ability_RIGHT_ON)) {
```

Comment:
it might not be, but is it possible to do: ``if(is_ability_active(Ability_RIGHT_ON) || (gBattleStatus.flags1 & 0x1000))`` here and ditch the other if statement entirely? If not, it's cool, just wondering.